### PR TITLE
python2 -> 3 for macOS 12.3

### PR DESCRIPTION
**Pull request by v-kat**
Resolves https://github.com/brave/brave-browser/issues/20335

https://developer.apple.com/documentation/macos-release-notes/macos-12_3-release-notes

Seemed to help me build on a Macbook pro M1

## Submitter Checklist:

- [X] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [ ] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [ ] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [ ] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [ ] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [ ] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

### CODE REVIEW

**Comment by v-kat** [comment]
I think CI might need `pylint3` but I'm not sure (the CI failure).

**Comment by mkarolin** [comment]
Unfortunately, transitioning from python2 to python3 in these scripts would require additional changes. For example, in python3 `print` arguments need to be in parentheses. CI pylint is using your change to apply python3 linting to the changed scripts, so all those pylint issues are brought on by the switch and need to be addressed. While the scripts haven't been upgraded you can install python2 on your MacOS manually, as indicated on https://github.com/brave/brave-browser/wiki/macOS-Development-Environment page:
<pre>
Python - 2.7 and 3+ - Note: On macOS 12.3 and higher Python 2.7 is no longer included by default and you will have to [install it manually](https://www.python.org/downloads/release/python-2718/)
</pre>

**Comment by v-kat** [comment]
Even with SIP on in MacOS 12.3 I'm getting:

```bash
➜  src git:(394e257d1ceb6) ✗ sudo ln -s /usr/bin/python3 /usr/bin/python 
Password:
ln: /usr/bin/python: Read-only file system
➜  src git:(394e257d1ceb6) ✗ ln -s /usr/bin/python3 /usr/bin/python 
ln: /usr/bin/python: Read-only file system
➜  src git:(394e257d1ceb6) ✗ sudo ln -sf /usr/bin/python3 /usr/bin/python
ln: /usr/bin/python: Read-only file system
```

I could maybe do a mount and workaround but it might be good to make the process simple and just bite the bullet and do the migration. I think even ubuntu is on 3 now (there's an easy apt workaround for that too?).
I have enough time that I could maybe work on this if you want...

**Comment by mkarolin** [comment]
Maybe folks who run into this already can recommend a solution: @kylehickinson @petemill

**Comment by kylehickinson** [comment]
I just reinstalled python2 via http://python.org/downloads/release/python-2718/

I believe @petemill uses `pyenv` and `python` ends up pointing to `python3` and everything works correctly

**Comment by bsclifton** [comment]
Ran into this too on my Mac Mini - I installed Python2 (thanks @kylehickinson):
http://python.org/downloads/release/python-2718/

And everything worked fine 👍 The python 2 => python 3 upgrade could take a while (for each of the scripts ☹️). But we should eventually do it (or capture an issue for that)

**Comment by mihaiplesa** [comment]
https://docs.python.org/3/library/2to3.html will also be useful. Today the uplift pipeline failed because it Python2 packages were not in the official repos anymore so we need to prioritize this.

File: script/apply-patches.py [hunk 1]
```diff
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
```
Comment (bridiver) [inline script/apply-patches.py:1]: I think these should all be vpython3

**Comment by bsclifton** [comment]
Closing as stale

**Pull request closed without merging**